src/main/kotlin/com/terraformation/backend/customer/db/DisclaimerStore.kt

```diff
@@ -0,0 +1,123 @@
+package com.terraformation.backend.customer.db
+
+import com.terraformation.backend.auth.currentUser
+import com.terraformation.backend.customer.model.DisclaimerModel
+import com.terraformation.backend.customer.model.requirePermissions
+import com.terraformation.backend.db.default_schema.DisclaimerId
+import com.terraformation.backend.db.default_schema.tables.daos.DisclaimersDao
+import com.terraformation.backend.db.default_schema.tables.daos.UserDisclaimersDao
+import com.terraformation.backend.db.default_schema.tables.pojos.DisclaimersRow
+import com.terraformation.backend.db.default_schema.tables.pojos.UserDisclaimersRow
+import com.terraformation.backend.db.default_schema.tables.references.DISCLAIMERS
+import com.terraformation.backend.db.default_schema.tables.references.USER_DISCLAIMERS
+import jakarta.inject.Named
+import java.time.Instant
+import java.time.InstantSource
+import org.jooq.DSLContext
+import org.jooq.impl.DSL
+
+@Named
+class DisclaimerStore(
+    private val clock: InstantSource,
+    private val disclaimersDao: DisclaimersDao,
+    private val dslContext: DSLContext,
+    private val userDisclaimersDao: UserDisclaimersDao,
+) {
+
+  fun createDisclaimer(content: String, effectiveOn: Instant): DisclaimerId {
+    requirePermissions { manageDisclaimers() }
+
+    val row =
+        DisclaimersRow(
+            content = content,
+            effectiveOn = effectiveOn,
+        )
+
+    disclaimersDao.insert(row)
+    return row.id!!
+  }
+
+  fun deleteDisclaimer(disclaimerId: DisclaimerId) {
+    requirePermissions { manageDisclaimers() }
+
+    disclaimersDao.deleteById(disclaimerId)
+  }
+
+  fun fetchAllDisclaimers(): List<DisclaimerModel> {
+    requirePermissions { manageDisclaimers() }
+
+    val usersMultiset =
+        with(USER_DISCLAIMERS) {
+          DSL.multiset(
+                  DSL.select(
+                          USER_ID,
+                          ACCEPTED_ON,
+                      )
+                      .from(this)
+                      .where(DISCLAIMER_ID.eq(DISCLAIMERS.ID)))
+              .convertFrom { result ->
+                result.associate { record -> record[USER_ID]!! to record[ACCEPTED_ON]!! }
+              }
+        }
+
+    return with(DISCLAIMERS) {
+      dslContext
+          .select(
+              ID,
+              CONTENT,
+              EFFECTIVE_ON,
+              usersMultiset,
+          )
+          .from(this)
+          .fetch { DisclaimerModel.of(it, usersMultiset) }
+    }
+  }
+
+  /**
+   * Fetch the latest effective disclaimer. Omits future disclaimers. Null if no effective
+   * disclaimer exists.
+   */
+  fun fetchCurrentDisclaimer(): DisclaimerModel? {
```

Comment:
If this returns null, will callers ever do anything other than throw an exception? If not, then maybe this should return `DisclaimerModel` and throw the exception itself.